Add Contributor ladder, contributor role, governance and roadmap document (#4)
metal3-io/community: CONTRIBUTOR-ROLE.md

# Contributor Roles

## Security Team

Security Team members are listed in the Metal3 user-guide under the
[Project Security Policy](https://book.metal3.io/security_policy#metal3-security-committee-members).
The security team is responsible for responding to all security issues reported
to Metal3 project. The team can be reached at `metal3-security@googlegroups.com`.
(Please do not report any security vulnerability to the committee members
directly.) The security team ensures that the issues are mitigated and
a responsible disclosure process is followed in a reasonable time. The security
team has a designated Security Lead. Currently,
[Tuomo Tanskanen](https://github.com/tuminoid) is acting as the Metal3 Security
Lead. Below are the responsibilities of the Security team.

> * Responsibilities:

Reviewer: This is still unclear if talking about the team or the lead, especially when some of the responsibilities are team's and some is lead's.

   * Thoroughly investigate all reports to the Security Team
   * Does not share any vulnerability information with others unless necessary
    to fix the issue

>    * Notify the reporter as the security issue moves through the identification

Reviewer: This is lead's task, to avoid any mixed communication.

    and resolution process
   * Uphold the [Project Security Policy](https://book.metal3.io/security_policy#metal3-security-committee-members).

* Qualifications:

   * Has a solid grasp of the overall architecture of Metal3, its usage in
    production environments, and its threat model.
   * Established the trust of Metal3 community members that they will uphold the
    security posture of Metal3

* Privileges:

   * Access to security reports
   * Membership of the security team mailing list
   * Security Lead has elevated access to Metal3 github organization through
   the `metal3_security_team` GitHub team.

The process of becoming a security team member is:

> 1. Nominated by a security team member and seconded by a maintainer

Reviewer: ```suggestion
1. Nominated by the Security Lead and seconded by an Metal3 Org Admin
```
I think any member along with any maintainer is not right.

Author: in that case we need to add another role Metal3 Org Admin. I wanted to avoid fine grained classification for now.

Reviewer: Good point. We have this role right now, so we need to document it as well.

Author: Would it be ok that I add an issue here and we discuss the criteria to be selected for that role within maintainers and add it later?

Reviewer: That role is not very relevant for a regular contributor, we can add it later. Will you incorporate the original comment anyways?

1. Shadowing an existing security team member

## CI Team

The CI team works to keep the CI infrastructure up and running. Please contact

> CI team at `metal3-ci@googlegroups.com`. (N.B. This group doesn't exist yet).

Reviewer: Not a fan of merging "does not exist", so let's get agreement that if we need such list or not, and then create the group and adjust the PR?

Author: Yes, the idea is to get some consensus here before merging. I will put it a hold so that it doenst merge. Opinion?

Reviewer 2: I agree with Tuomo on this, we should create the group first

* Responsibilities:

   * Keeping CI infrastructure healthy
   * Investigating CI infrastructure problems
   * Provide visibility into CI spend and status

* Qualifications:

   * Must be a Metal3 Org member
   * A basic understanding of how the CI system works

* Privileges:

>    * Privileged access rights to configuring CI infrastructure is available

Reviewer: We should make it clear what privileges you actually get. This isn't very helpful right now. CI access gives you really broad rights, as you need GH side rights, Jenkins rights, Prow rights?

If we are not ready for open CI team, then maybe drop it from the document until we are?

Author: I want to keep it because its a large part of work. We should think how to make it open towards any org member.

Reviewer: OK. Let's then make it explicit that it is work in progress, and not just mailing list missing?

Author: I am not intending to make this PR hang on this. We can keep updating the process later on.

Reviewer: No need to hang, agreed. Was just suggesting adding explicit note that opening the CI role is work in progress, so its clear.

   to a small subset of people currently.

## Release Team

The Release team is responsible for releasing minor and patch releases from
different repositories in Metal3. Please contact Release team at
`metal3-release@googlegroups.com`. (N.B. This group doesn't exist yet).

* Responsibilities:

>    * Making sure the repositories are ready to be released.

Reviewer: ```suggestion
   * Making sure the repositories are ready to be released
```

>    * Performing minor and patch releases.

Reviewer: ```suggestion
   * Performing minor and patch releases
```

   * Communicate the releases to the community

* Qualifications:

   * Must be a Metal3 Org member

>    * A proper understanding of release process of different Metal3 repositories.

Reviewer: ```suggestion
   * A proper understanding of release process of different Metal3 repositories
```

* Privileges:

>    * Release team requires elevated access to Metal3 github organization through

Reviewer: ```suggestion
   * Release team requires elevated access to Metal3 GitHub organization through
```

>    the `metal3_release_team` github team.

Reviewer: ```suggestion
   the `metal3_release_team` GitHub team.
```

The process of becoming a release team member is:

> 1. Nominated by a community member and seconded by a maintainer

Reviewer: Again, very loose definition. Org Admin would make more sense IMO, since that is required to add the person to the team as well.

Author: check my previous comment.

1. Shadowing an existing release team member

## Maintainers

[Maintainers](maintainers/ALL-OWNERS) are very established contributors who are
responsible for the entire project. Detailed responsibilities and guideline to

> become a maintainer is documented [here](maintainers/README.md).

Reviewer: Instead of pointing to that readme, it should point to CONTRIBUTOR-LADDER.md

Author: actually I wanted to point to README.

Reviewer: OK, in that case can we more refer to the README? If we create second copy of the same info, but a bit different. Having it in multiple places and especially having content differ just creates confusion.

Author: Yes I can agree to that. I thought of referring the section here but then moved on to put a concise version here to make it consistent with other roles of this document. Do you want me to remove everything I wrote here and just refer to the README?

Reviewer: IMO not remove everything, leave the responsibilities and privileges part so it is consistent with other roles, but replace the "becoming approver" part with a link to README, since README is all about becoming approver.

   * Security Lead has elevated access to Metal3 github organization through
   the `metal3_security_team` GitHub team.

The process of becoming a security team member is:

1. Nominated by a security team member and by an Metal3 Org Admin.
1. Shadowing an existing security team member

## CI Team

The CI team works to keep the CI infrastructure up and running. Please contact
CI team at `metal3-ci@googlegroups.com`. (N.B. This group doesn't exist yet).

* Responsibilities:

   * Keeping CI infrastructure healthy
   * Investigating CI infrastructure problems
   * Provide visibility into CI spend and status

* Qualifications:

   * Must be a Metal3 Org member
   * A basic understanding of how the CI system works

* Privileges:

   * Privileged access rights to configuring CI infrastructure is available
   to a small subset of people currently. The process definition of becoming a
   CI Team member is still work in progress.

## Release Team

The Release team is responsible for releasing minor and patch releases from
different repositories in Metal3. Please contact Release team at
`metal3-release@googlegroups.com`. (N.B. This group doesn't exist yet).

> 

Reviewer 2: same here

Author: Cool, since we agree, We can create those and I will update PR afterwards
